src/parse.ros2idl.test.ts

```diff
@@ -0,0 +1,886 @@
+      // All of these comments should be ignored
+      module action {
+        /** another comment */
+        module MyAction_Goal_Constants {
+          // two here of the same type
+          // _another one_
+          const string tricky = "/** is this a comment? */ // hopefully not";
+        };
+        /** a multi
+         * line
+         * comment
+         */
+        struct MyAction_Goal {
+          // two here of different types
+          /** maybe one more that */
+          int32 input_value;
+        };
+      };
+    `;
+    const types = parse(msgDef, { ros2idl: true });
+    expect(types).toEqual([
+      {
+        definitions: [
+          {
+            isConstant: true,
+            value: "/** is this a comment? */ // hopefully not",
+            valueText: "/** is this a comment? */ // hopefully not",
+            upperBound: undefined,
+            isComplex: false,
+            name: "tricky",
+            type: "string",
+          },
+        ],
+        name: "action::MyAction_Goal_Constants",
+      },
+      {
+        definitions: [
+          {
+            isComplex: false,
+            name: "input_value",
+            type: "int32",
+          },
+        ],
+        name: "action::MyAction_Goal",
+      },
+    ]);
+  });
+  it("can parse multiple forward declarations on same line with default annotation", () => {
+    const msgDef = ` 
+    module action {
+      struct MyAction_Goal {
+        @default(value=5)
+        int32 int1, int2;
+      };
+    };
+    `;
+    const types = parse(msgDef, { ros2idl: true });
+    expect(types).toEqual([
+      {
+        definitions: [
+          {
+            defaultValue: 5,
+            isComplex: false,
+            name: "int1",
+            type: "int32",
+          },
+          {
+            defaultValue: 5,
+            isComplex: false,
+            name: "int2",
+            type: "int32",
+          },
+        ],
+        name: "action::MyAction_Goal",
+      },
+    ]);
+  });
+  /****************  Not supported by IDL (as far as I can tell) */
+  it("cannot parse multiple const declarations in a single line", () => {
+    const msgDef = ` 
+      module action {
+        module MyAction_Goal_Constants {
+          const short short1, short2 = -23;
+        };
+      };
+    `;
+    expect(() => parse(msgDef, { ros2idl: true })).toThrow(/unexpected , token: ","/i);
+  });
+  /****************  Syntax Errors */
+  it("missing bracket at the end will result in end of input error", () => {
+    const msgDef = ` 
+    module rosidl_parser {
+      module action {
+        module MyAction_Goal_Constants {
+          const short SHORT_CONSTANT = -23;
+        };
+        struct MyAction_Goal {
+          int32 input_value;
+        };
+      };
+    `;
+    expect(() => parse(msgDef, { ros2idl: true })).toThrow(
+      `Could not parse message definition (unexpected end of input): '${msgDef}'`,
+    );
+  });
+  it("cannot parse empty module", () => {
```

Comment:
is this actually a syntax error in IDL or just something we don't support today?

Reply:
from IDL spec: 
> <module_dcl> ::= "module" <identifier> "{" <definition>+ "}"